src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
 
+    private static final int INITIAL_SIZE = 10;
+
+    private Object[] array = new Object[INITIAL_SIZE];
+    private int numberOfElements;
+
     @Override
     public void add(T value) {
-
+        checkWhetherResize();
+        array[numberOfElements++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > numberOfElements) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+        checkWhetherResize();
+        System.arraycopy(array, index, array, index + 1, numberOfElements - index);
+        numberOfElements++;
+        array[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        exceptionCheck(index);
+        return (T) array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        exceptionCheck(index);
+        array[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        exceptionCheck(index);
+        Object removedElement = array[index];
+        System.arraycopy(array, index + 1, array, index, numberOfElements - index);
+        numberOfElements--;
+        return (T) removedElement;
     }
 
     @Override
     public T remove(T t) {
-        return null;
+        Object removedElement;
+        for (int i = 0; i < numberOfElements; i++) {
+            if (array[i] == null ? array[i] == t : array[i].equals(t)) {
+                removedElement = array[i];
+                remove(i);
+                return (T) removedElement;
+            }
+        }
+        throw new NoSuchElementException("There is no elements with this value");
     }
 
     @Override
     public int size() {
-        return 0;
+        return numberOfElements;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        for (int i = 0; i < numberOfElements; i++) {
+            if (array[i] != null) {
+                return false;
+            }
+        }
+        return true;
```

Comment:
let's simplify this
we have a field that stores current number of elements in array
if it is 0, we should return true

Reply:
done